site/src/ThemeTypeLayer.jsx

```diff
@@ -0,0 +1,211 @@
+import { Layer } from "react-map-gl/maplibre";
+import PropTypes from "prop-types";
+import "./CustomControls.css";
+
+const colorExpression = (color) => {
+  return [
+    "case",
+    ["boolean", ["feature-state", "selected"], false],
+    "white",
+    color,
+  ];
+};
+
+const ThemeTypeLayer = ({
```

Comment:
Refactored into separate file. I made the following changes:
1. "active" layer styling
2. Support for minzoom
3. Outline layers for polygons